forte/processors/data_augment/algorithms/word_splitting_processor.py

```diff
@@ -0,0 +1,126 @@
+
+    def _augment(self, input_pack: MultiPack, aug_pack_names: List[str]):
+        r"""
+        This function splits a given word at a random position and replaces
+        the original word with 2 split parts of it.
+        """
+        augment_entry = get_class(self.configs["augment_entry"])
+
+        for pack_name in aug_pack_names:
+            data_pack: DataPack = input_pack.get_pack(pack_name)
+
+            annotations: List[Annotation] = []
+            indexes: List[int] = []
+            endings = []
+            annos: Iterable[Annotation] = data_pack.get(augment_entry)
+            for idx, anno in enumerate(annos):
+                annotations.append(anno)
+                indexes.append(idx)
+                endings.append(anno.end)
+            if len(annotations) > 0:
+                annotation_to_split = random.sample(
+                    [
+                        (anno, idx)
+                        for (anno, idx) in zip(annotations, indexes)
+                        if (anno.end - anno.begin) > 1
+                    ],
+                    ceil(self.configs["alpha"] * len(annotations)),
+                )
+                annotation_to_split = sorted(
+                    annotation_to_split, key=lambda x: x[1], reverse=True
+                )
+                for curr_anno in annotation_to_split:
+                    src_anno, src_idx = curr_anno
+                    splitting_position = random.randrange(
+                        1, (src_anno.end - src_anno.begin)
+                    )
+                    word_split = [
+                        src_anno.text[:splitting_position],
+                        src_anno.text[splitting_position:],
+                    ]
+                    if src_idx != 0:
+                        first_position = endings[src_idx - 1] + 1
+                        second_position = endings[src_idx]
+                        word_split[1] = " " + word_split[1]
+                    else:
+                        first_position = 0
+                        second_position = endings[0]
+                        word_split[1] = " " + word_split[1]
+
+                    self._insert(word_split[1], data_pack, second_position)
+                    self._delete(src_anno)
+                    self._insert(word_split[0], data_pack, first_position)
+
+    @classmethod
+    def default_configs(cls):
+        """
+        Returns:
+            A dictionary with the default config for this processor.
+            Additional keys for determining how many words will be split:
+            - alpha: 0 <= alpha <= 1. indicates the percent of the words
+            in a sentence that are changed.
+        Config Values:
+            - `alpha` (float): The probability of splitting in [0, 1](Default = 0.1).
+        """
+        config = super().default_configs()
+        config.update(
+            {
+                "augment_entry": "ft.onto.base_ontology.Token",
+                "other_entry_policy": {
```

Comment:
Is the "type" key under this "other_entry_policy" used?

Btw, I think the docstring should also explain what other keys are, especially what the "other_entry_policy" does and how to use it.

And I have a few suggestions for the default values:

1. `ft.onto.base_ontology.EntityMention` does not seem to be important and be the default. "ft.onto.base_ontology.Document" and "Sentence" might be fine.

Reply:
I had gotten a configuration error on removing "type"

Reply:
Understood.

Reply:
Just update the other docstrings and I think it is fine. 

I think "type" might be an obsolete config but not fully erased from the base implementation. Would you mind find out what it does and see what we should do with it?